File: src/Algorithms/ConstructiveAlgorithms/medium/NewYearChaos.java
package Algorithms.ConstructiveAlgorithms.medium;

import java.util.*;

public class NewYearChaos {

    public static void main(String [] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        while (t-- > 0) {
            int n = in.nextInt();
            int [] ar = new int[n];
            for (int i=0;i<n;i++)
                ar[i] = in.nextInt();
            findResult(ar);
        }
        in.close();

    }

    public static void findResult(int [] ar) {
        int numOfBribes = 0;
        int temp;
        for (int i=ar.length-1;i>=2;i--) {
            if (ar[i] == i+1) {
                //correct
            }else if (ar[i-1] == i+1) {
                temp = ar[i-1];
                ar[i-1] = ar[i];
                ar[i] = temp;

                numOfBribes++;
            }else if (ar[i-2] == i+1) {
                temp = ar[i-2];
                ar[i-2] = ar[i-1];
                ar[i-1] = temp;

                temp = ar[i-1];
                ar[i-1] = ar[i];
                ar[i] = temp;

                numOfBribes+= 2;
            }else {
                System.out.println("Too chaotic");
                return;
            }
        }
        if (ar[0] != 1)
            numOfBribes+= 1;
        System.out.println(numOfBribes);
    }

}

File: src/TheHackerRankInterviewPreparationKit/Arrays/NewYearChaos.kt
package TheHackerRankInterviewPreparationKit.Arrays
//Date 1-8-2018
import java.util.*
fun main (args: Array<String>){
    val s = Scanner(System.`in`)
    val t = s.nextInt()
    for ( testcase in 1 .. t){
        val n = s.nextInt()
        val arr = IntArray(n)
        for(i in 0 until n)
            arr[i]=s.nextInt()
        var counter = 0
        var isChaotic = false
        for ( i in n-1 downTo 0)
        {
            if(arr[i]- (i+1) >2)
            {
                isChaotic=true
            }
            for ( j in Math.max(0,arr[i]-2) .. i)
                if (arr[j] > arr[i])
                    counter++
        }
        if(isChaotic)
            println("Too chaotic")
        else
            println(counter)
    }
    s.close()
}